libs/api/domains/license-service/src/lib/mappers/identityDocumentMapper.ts

```diff
@@ -0,0 +1,285 @@
+import { Locale } from '@island.is/shared/types'
+import { LICENSE_NAMESPACE } from '../licenseService.constants'
+import { Injectable } from '@nestjs/common'
+import { isDefined } from '@island.is/shared/utils'
+import {
+  type IdentityDocument,
+  type IdentityDocumentChild,
+} from '@island.is/clients/passports'
+import { FormatMessage, IntlService } from '@island.is/cms-translations'
+import { m } from '../messages'
+import { GenericUserLicenseAlert } from '../dto/GenericUserLicenseAlert.dto'
+import { GenericUserLicenseMetaLinks } from '../dto/GenericUserLicenseMetaLinks.dto'
+import { GenericUserLicenseMetaTag } from '../dto/GenericUserLicenseMetaTag.dto'
+import { capitalizeEveryWord } from '../utils/capitalize'
+import { Payload } from '../dto/Payload.dto'
+import { formatDate } from '../utils'
+import {
+  AlertType,
+  ExpiryStatus,
+  GenericLicenseDataFieldType,
+  GenericLicenseMappedPayloadResponse,
+  GenericLicenseMapper,
+  GenericUserLicenseDataFieldTagColor,
+  GenericUserLicenseDataFieldTagType,
+  GenericUserLicenseMetaLinksType,
+} from '../licenceService.type'
+import { GenericLicenseDataField } from '../dto/GenericLicenseDataField.dto'
+
+const isChildDocument = (
+  passport: IdentityDocument | IdentityDocumentChild,
+): passport is IdentityDocumentChild => {
+  return (passport as IdentityDocumentChild).childNationalId !== undefined
+}
+
+@Injectable()
+export class IdentityDocumentMapper implements GenericLicenseMapper {
+  constructor(private readonly intlService: IntlService) {}
+  async parsePayload(
+    payload: Array<unknown>,
+    locale: Locale = 'is',
+  ): Promise<Array<GenericLicenseMappedPayloadResponse>> {
+    const { formatMessage } = await this.intlService.useIntl(
+      [LICENSE_NAMESPACE],
+      locale,
+    )
+
+    const emptyIdentityDocument = {
```

Comment:
Will I always get an empty license if I don't have an identity document, even if I have a passport?

Reply:
yea, they specified that the identity document should have **exactly** the same ui logic as the passport